File: languages/js/src/entity/availability/ipns_loader.ts
import * as proto from "../../bridge/proto/bloock_availability_entities";
import { Loader } from "./loader";
import { LoaderArgs } from "./loader_args";

/**
 * Represents a loader for IPNS data availability.
 */
export class IpnsLoader implements Loader {
  type: proto.DataAvailabilityType;
  args: LoaderArgs;

  /**
   * Constructs a IpnsLoader object with the specified parameters.
   * @param id
   */
  constructor(id: string) {
    this.type = proto.DataAvailabilityType.IPNS;
    this.args = new LoaderArgs(id);
  }

  public toProto(): proto.Loader {
    return proto.Loader.fromPartial({
      type: this.type,
      args: this.args.toProto()
    });
  }
}

File: languages/go/entity/availability/ipns_loader.go
package availability

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// IpnsLoader represents a loader for IPNS data availability.
type IpnsLoader struct {
	Type proto.DataAvailabilityType
	Args LoaderArgs
}

// NewIpnsLoader creates an IpnsLoader instance with the provided IPNS hash.
func NewIpnsLoader(hash string) IpfsLoader {
	return IpfsLoader{
		Type: proto.DataAvailabilityType_IPNS,
		Args: LoaderArgs{
			hash,
		},
	}
}

func (e IpnsLoader) ToProto() *proto.Loader {
	return &proto.Loader{
		Type: e.Type,
		Args: e.Args.ToProto(),
	}
}
